Automattic/jetpack
# Story block: Prevent 'key is not a prop' warning in ProgressBullet component

Fixes https://github.com/Automattic/jetpack/issues/39805

## Proposed changes:

* This PR adds a new prop, 'bulletIndex', to the 'ProgressBullet' component, to pass through the index in order to create a value for the key prop within that component. This is because 'key' itself is an internal React prop, so we are not allowed to pass it to another component as a prop.

### Other information:


## Jetpack product discussion

https://github.com/Automattic/jetpack/issues/39805

## Does this pull request change what data or activity we track or use?

No.

## Testing instructions:

To replicate the issue:

* Make sure you have either Gutenberg trunk, or the WordPress beta tester plugin running the latest nightly.
* Go to Appearance > Site Editor
* Click on the Style icon in the top right corner (you may need to click on the page for the icons to display)
* Click on the Eye icon in the Styles sidebar.
* Click on the Media tab to preview the Slideshow block (amongst other blocks)
* You'll notice various console warnings, including one related to the Story block: `Warning: ProgressBullet: 'key' is not a prop. Trying to access it will result in 'undefined' being returned. If you need to access the same value within the child component, you should pass it as a different prop.`

To test the fix:

* Apply this PR and follow the above steps again. Those warnings related to the Story block should be gone.
